### DISCUSSION TITLE
No failure on running all tests, consistent failures on running just one/set?

### DISCUSSION THREAD

**Discussion by actual-nh** [CleverRaven/Cataclysm-DDA#46404] [Q&A] [answered]
Is it at all expected (e.g., skipping tests due to time - I'm not very familiar with Catch) that running `make check` does not show failures, but `tests/cata_test weary_recovery --rng-seed [same seed as make check]` shows at least one failed assertion (and thus test), consistently? Checking before I submit yet another test-related issue...

This is even if I do `make clean-tests`, `make tests`, and only then do another single-test `cata_test` invocation. (See #46384 for OS, etc information.)

Oh. Ping: @anothersimulacrum. Forgot, sorry! (@Qrox, any ideas?)

**Comment by Qrox** [comment, marked as answer]
It could be that when you ran all the tests, some test before the weary recovery test changed the game states so the weary recovery test ran differently. Without debugging it's hard to say what exactly caused the problem though.

**Reply by actual-nh** [reply]
Sigh... that's what I was afraid of. Thanks!

**Reply by anothersimulacrum** [reply]
You can try adding `clear_map()` and `clear_avatar()` with some degree of liberty throughout the tests and it may fix it.

**Reply by actual-nh** [reply]
OK. I am also planning a PR to add `--order rand` to `make check`. (Current status: If one runs `weary_*`, no errors. Without `weary_assorted_tasks` being first (including with `--order rand`), errors.) Not sure how much further I'll get tonight - hope can do tomorrow. (I will have considerably less time to work on this after Jan 1 or so - new semester starting soon.)

**Reply by actual-nh** [reply]
> You can try adding `clear_map()` and `clear_avatar()` with some degree of liberty throughout the tests and it may fix it.

@anothersimulacrum: Unfortunately, do_activity already does both of those.

**Comment by actual-nh** [comment]
There are definitely some clear_map() and clear_avatar() additions needed, from running a check using --order rand. Among them is that the avatar frequently starves to death during the weary tests...

To add to that, backtrace problems - #46439.
